multimodal/src/autogluon/multimodal/models/mmocr_text_detection.py

```diff
@@ -0,0 +1,156 @@
+import logging
+import warnings
+from typing import Optional
+
+import torch
+from mim.commands.download import download
+from torch import nn
+
+try:
+    import mmcv
+    from mmcv.parallel import scatter
+    from mmcv.runner import load_checkpoint
+except ImportError:
+    mmcv = None
+
+try:
+    import mmocr
+    from mmocr.models import build_detector
+    from mmocr.utils.model import revert_sync_batchnorm
+except ImportError:
+    mmocr = None
+
+try:
+    import mmdet
+    from mmdet.core import get_classes
+except ImportError:
+    mmdet = None
+
+from ..constants import AUTOMM, BBOX, COLUMN, COLUMN_FEATURES, FEATURES, IMAGE, IMAGE_VALID_NUM, LABEL, LOGITS, MASKS
+from .utils import assign_layer_ids, get_column_features, get_model_head
+
+logger = logging.getLogger(AUTOMM)
+
+
+class MMOCRAutoModelForTextDetection(nn.Module):
+    """
+    Support MMOCR object detection models.
+    Refer to https://github.com/open-mmlab/mmocr
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        checkpoint_name: str,
+        num_classes: Optional[int] = None,
+        pretrained: Optional[bool] = True,
+    ):
+        """
+        Load a pretrained ocr text detection detector from MMOCR.
+
+        Parameters
+        ----------
+        prefix
+            The prefix of the MMdetAutoModelForTextDetection model.
+        checkpoint_name
+            Name of the mmdet checkpoint.
+        num_classes
+            The number of classes.
+        pretrained
+            Whether using the pretrained mmdet models. If pretrained=True, download the pretrained model.
+        """
+        super().__init__()
+        logger.debug(f"initializing {checkpoint_name}")
+        self.checkpoint_name = checkpoint_name
+        self.pretrained = pretrained
+
+        # download config and checkpoint files using openmim
+        checkpoints = download(package="mmocr", configs=[checkpoint_name], dest_root=".")
+
+        # read config files
+        assert mmcv is not None, "Please install mmcv-full by: mim install mmcv-full."
+        config_file = checkpoint_name + ".py"
+        if isinstance(config_file, str):
+            self.config = mmcv.Config.fromfile(config_file)
+
+        # build model and load pretrained weights
+        assert mmocr is not None, "Please install MMOCR by: pip install mmocr."
+
+        checkpoint = checkpoints[0]
+        self.model = build_detector(self.config.model, test_cfg=self.config.get("test_cfg"))
+        if checkpoint is not None:
+            checkpoint = load_checkpoint(self.model, checkpoint, map_location="cpu")
```

Comment:
This part is similar to the part in `MMDetAutoModelForObjectDetection` since they all use the `mmcv` logics. Consider wrap them into a util and put in `utils.py`.

Reply:
The `build_detector` function is not sharable, only line 67 to line 74 can be shared. Maybe we can keep it this way for now, it's easier to modify. Later, if all CV tasks in mmlab can reuse this part, we can wrap them into a util.

Reply:
Sure. Good to mark a TODO.

Reply:
Yes, please add a TODO here in case we forget, thanks.